### PR TITLE
aws-cli: update to 1.22.23 (with deps)

### PR DESCRIPTION
**Pull request by bobertlo**
#### Testing the changes
- I tested the changes in this PR: yes

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc

### CODE REVIEW

**Comment by uhohspaghetios** [comment]
Works for me.

**Comment by dotnetfox** [comment]
@bobertlo, your particular comment in all these commits caught my interest.

```sh
# pytest marked deprecated, suggests using tox (which does not test against system packages)
```

You see, I too got entrapped with tests (PR #34489) and it seems I have to explicitly specify `python3-pytest` in `checkdepends` along with packages it uses during tests.

If `python3-pytest` is not defined, `do_check` falls back to that "deprecated testing" which you mentioned.

```sh
do_check() {
    if python3 -c 'import pytest' >/dev/null 2>&1; then
        PYTHONPATH="$(cd build/lib* && pwd)" \
            python3 -m pytest ${make_check_args} ${make_check_target}
    else
        # Fall back to deprecated setup.py test orchestration without pytest
        if [ -z "$make_check_target" ]; then
            if ! python3 setup.py --help test >/dev/null 2>&1; then
                msg_warn "No command 'test' defined by setup.py.\n"
                return 0
            fi
        fi

        : ${make_check_target:=test}
        python3 setup.py ${make_check_target} ${make_check_args}
    fi
}
```

To be certain, the `python3-module` build style does not append `python3-pytest` to anywhere.

As you can see above, it is mentioned only within `do_check`.

Hope this information will be useful for you.

**Comment by paper42** [comment]
> ```
> # pytest marked deprecated, suggests using tox (which does not test against system packages)
> ```

Where exactly is this comment from? They migrated to pytest in september, so I would assume it's not already deprecated.

**Comment by bobertlo** [comment]
> > ```
> > # pytest marked deprecated, suggests using tox (which does not test against system packages)
> > ```
> 
> Where exactly is this comment from? They migrated to pytest in september, so I would assume it's not already deprecated.

When I got the tests running with their deps installed and pytest it still says:

```
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox
```

The Tox tests install their own python3-venv and download package dependencies with pip.

@dotnetfox the tests will run but give that warning message and I did find that code you mentioned as well as figure out all the dependencies. I think they have some proprietary secret sauce going on with these tests. I got the test failures down to like 300-ish (out of ~30,000) and they were all saying invalid AWS credentials.

I spent a decent chunk of time on two different nights trying to get these tests to run.

**Comment by paper42** [comment]
> When I got the tests running with their deps installed and pytest it still says:
> 
> ```
> WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox
> ```

That's a message from `setup.py test` which is deprecated. We should use pytest if possible:
```sh
checkdepends="python3-pytest $depends"
```
But it seems like there is a lot of tests and we will need to skip some of them or skip them anyway. Even if you decide it's not worth fixing them (which I think might be ok in this case), the reason for this should be modified.

**Comment by bobertlo** [comment]
@paper42 oops I guess I misunderstood that. sorry!

I had to disable all the intergration tests. They seem to want aws credentials and have other issues apparently related to the build environment.

File: srcpkgs/python3-botocore/template [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 pkgname=python3-botocore
-version=1.17.28
-revision=4
+version=1.23.23
+revision=1
 wrksrc="botocore-${version}"
 build_style=python3-module
 hostmakedepends="python3-setuptools"
 depends="python3-dateutil python3-docutils python3-jmespath python3-urllib3"
+checkdepends="python3-dateutil python3-docutils python3-jmespath python3-urllib3 python3-pytest python3-jsonschema"
```
Comment (paper42) [inline srcpkgs/python3-botocore/template:9]:
```suggestion
checkdepends="$depends python3-pytest python3-jsonschema"
```

File: srcpkgs/python3-botocore/template [hunk 2]
```diff
@@ -1,14 +1,21 @@
 # Template file for 'python3-botocore'
 pkgname=python3-botocore
-version=1.17.28
-revision=4
+version=1.23.23
+revision=1
 wrksrc="botocore-${version}"
 build_style=python3-module
 hostmakedepends="python3-setuptools"
 depends="python3-dateutil python3-docutils python3-jmespath python3-urllib3"
```
Comment (paper42) [inline srcpkgs/python3-botocore/template:8]: https://github.com/boto/botocore/blob/23ee17f5446c78167ff442302471f9928c3b4b7c/.changes/1.18.0.json#L32

```json
  {
    "category": "dependency", 
    "description": "botocore has removed docutils as a required dependency", 
    "type": "feature"
  }
```

File: srcpkgs/python3-botocore/template [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
 homepage="https://github.com/boto/botocore"
 distfiles="https://github.com/boto/botocore/archive/${version}.tar.gz"
-checksum=c9a4b4ffedeff89848a899bd4634be46a21a190690b505c2050888c0a5447936
+checksum=6b23a1e91387c1a461f10b17bde9ace15b3c128ed385646ab058644dfedd2c47
+
+do_check() {
+	# integration tests want aws credentials
+	rm -r tests/functional/leak # these 6 tests fail, probably fixable
+	pytest tests/functional tests/unit
+}
```
Comment (paper42) [inline srcpkgs/python3-botocore/template:21]:
```suggestion
# integration tests want aws credentials
make_check_target="tests/functional tests/unit"

pre_check() {
	rm -r tests/functional/leak # these 6 tests fail, probably fixable
}
```

**Pull request merged**